11/umbraco-cms/tutorials/add-google-authentication.md

```diff
@@ -28,249 +28,307 @@ I'm sure a lot of content editors and implementors of your Umbraco sites would l
+The NuGet Package Manager Console lets you use NuGet PowerShell commands to manage NuGet packages on your project. You can use this option if you are comfortable using the Package Manager Console (PowerShell). The command listed below is specific to the Package Manager Console in Visual Studio:
 
 1. Open your project/solution in Visual Studio.
-2. Go to **Tools** -> **NuGet Package Manager** -> **Package Manager Console**. A package manager console appears at the bottom where you can install packages with commands. In this console, type the following:
+2. Go to **Tools** > **NuGet Package Manager** > **Package Manager Console**.
+
+A package manager console appears at the bottom where you can install packages with commands.
+
+3. Type the following in the console:
 
 ```js
 Install-Package Microsoft.AspNetCore.Authentication.Google -Version 5.0.0
 ```
 
 #### Option 2: NuGet Package Manager
 
-The NuGet Package Manager UI in Visual Studio on Windows allows you to install, uninstall, and update NuGet packages in projects and solutions.
-
-1. Go to **Tools** -> **NuGet Package Manager** -> **Manage NuGet Packages for Solution**.
-2. In the Browse tab, type `Microsoft.AspNetCore.Authentication.Google` in the search field.
-3. Select the **version** from the drop-down and click **Install**.
+The NuGet Package Manager UI in Visual Studio allows you to manage NuGet packages in projects and solutions.
 
-<figure><img src="../../../10/umbraco-cms/tutorials/images/Install_Package.png" alt=""><figcaption></figcaption></figure>
+1. Open your project in Visual Studio.
+2. Go to **Tools** -> **NuGet Package Manager** -> **Manage NuGet Packages for Solution**.
+3. Select the **Browse** tab.
+4. Type `Microsoft.AspNetCore.Authentication.Google` in the search field.
+5. Ensure that your project is checked.
+6. Select the **version** from the drop-down and click **Install**.
 
 For more information on installing and managing packages in Visual Studio, see [Microsoft Documentation](https://docs.microsoft.com/en-us/nuget/consume-packages/install-use-packages-visual-studio).
 
-## Configuring the solution to allow Google logins
+## 3. Configuring the solution to allow Google logins
+
+To use an external login provider such as Google on your Umbraco CMS project, you have to implement a couple of new classes:
 
-1.  To enable a user to link their user account to an external login provider such as Google in the Umbraco Backoffice, you have to implement a custom-named configuration `BackOfficeExternalLoginProviderOptions` for users.
+* A custom-named configuration class.
+* A static extension class.
 
-    You can create a `GoogleBackOfficeExternalLoginProviderOptions.cs` file in a location of your choice.\
-    \
-    For tutorial purposes, I have created the file in `App_Code/Google_Authentication` folder.
+You can create these files in a location of your choice. In this tutorial, the files will be added to an `App_Code/Google_Authentication` folder.
```

Comment:
I must admit that I'm really not a fan of the `App_Code/Google_Authentication` folder name, `App_Code` was very much a .Net Framework thing, and is not a thing more in modern .Net, additionally using underscores in namespaces is generally not recommended.

Maybe something along the `ExternalUserLogins/Google` or something would be better. (Do note that the namespace in the code would have to change as well)